plugin/manager.go

```diff
@@ -84,6 +82,77 @@ func (m *CLIManager) List(ctx context.Context) ([]string, error) {
 	return plugins, nil
 }
 
+// Install installs a plugin at filePath to the system.
+//
+// It returns the new plugin metadata and a nil eror if and only if
+// the installation succeeded.
+//
+// If plugin already exists, and overwrite is not set, then the new plugin
+// version MUST be higher than the existing plugin version.
+// If overwrite is set, version check is skipped.
+// On sucess, existing plugin metadata and new plugin metadata are returned.
+//
+// If plugin does not exist, directly install the plugin from filePath.
+// On success, the new plugin metadata is returned.
+func (m *CLIManager) Install(ctx context.Context, filePath string, overwrite bool) (*proto.GetMetadataResponse, *proto.GetMetadataResponse, error) {
+	// validate and get new plugin metadata
+	pluginName, err := ExtractPluginNameFromFileName(filepath.Base(filePath))
```

Comment:
Because both `ExtractPluginNameFromFileName` and `validatePluginFileExtensionAgainstOS` need the file name rather than the entire path, can we extract the file name and then pass to both functions.
```suggestion
        pluginFile := filepath.Base(filePath)
	pluginName, err := ExtractPluginNameFromFileName(pluginFile)
	......
	if err := validatePluginFileExtensionAgainstOS(pluginFile, pluginName); err != nil {
```
Then we can update `validatePluginFileExtensionAgainstOS` to remove plugin.go: line 274